devenv/dev-dashboards/all-panels.json

```diff
@@ -0,0 +1,1122 @@
+                "tooltip": false,
+                "viz": false
+              },
+              "lineWidth": 1
+            },
+            "links": [
+              {
+                "title": "Data link",
+                "url": "/"
+              }
+            ],
+            "mappings": [],
+            "thresholds": {
+              "mode": "absolute",
+              "steps": [
+                {
+                  "color": "green"
+                },
+                {
+                  "color": "red",
+                  "value": 80
+                }
+              ]
+            }
+          },
+          "overrides": []
+        },
+        "gridPos": {
+          "h": 8,
+          "w": 12,
+          "x": 12,
+          "y": 13
+        },
+        "id": 4,
+        "options": {
+          "bucketOffset": 0,
+          "combine": false,
+          "legend": {
+            "calcs": [],
+            "displayMode": "list",
+            "placement": "bottom"
+          }
+        },
+        "title": "Histogram",
+        "type": "histogram"
+      },
+      {
+        "gridPos": {
+          "h": 8,
+          "w": 12,
+          "x": 0,
+          "y": 21
+        },
+        "id": 28,
+        "options": {
+          "dedupStrategy": "none",
+          "enableLogDetails": true,
+          "prettifyLogMessage": false,
+          "showCommonLabels": false,
+          "showLabels": false,
+          "showTime": false,
+          "sortOrder": "Descending",
+          "wrapLogMessage": false
+        },
+        "targets": [
+          {
+            "refId": "A",
+            "scenarioId": "logs"
+          }
+        ],
+        "title": "Logs",
+        "type": "logs"
+      },
+      {
+        "gridPos": {
+          "h": 8,
+          "w": 12,
+          "x": 12,
+          "y": 21
+        },
+        "id": 8,
+        "options": {
+          "maxItems": 10,
+          "query": "",
+          "showHeadings": true,
+          "showRecentlyViewed": true,
+          "showSearch": true,
+          "showStarred": true,
+          "tags": []
+        },
+        "pluginVersion": "8.1.0-pre",
+        "title": "Dashboard list",
+        "type": "dashlist"
+      },
+      {
+        "gridPos": {
+          "h": 8,
+          "w": 12,
+          "x": 0,
+          "y": 29
+        },
+        "id": 30,
+        "title": "Panel list",
+        "type": "pluginlist"
+      },
+      {
+        "gridPos": {
+          "h": 8,
+          "w": 12,
+          "x": 12,
+          "y": 29
+        },
+        "id": 6,
+        "options": {
+          "alertName": "",
+          "dashboardAlerts": false,
+          "maxItems": 20,
+          "showInstances": false,
+          "sortOrder": 1,
+          "stateFilter": {
+            "firing": true,
+            "inactive": false,
+            "pending": true
+          }
+        },
+        "title": "Alert list",
+        "type": "alertlist"
+      },
+      {
+        "cards": {
+        },
```

Comment:
This looks like a proper schema violation - heatmap isn't schematized, but this is within the segment of the structure that's owned by the core schema. Either we need to add something for `cards`, or refactor the way that heatmap emits data.

i'm also quite concerned that there are several other values in inappropriate places, here - we made a hack for the old Graph panel, but i really don't want to do another. Ideally, we'd refactor to ensure that heatmap stays within the boundaries of where panel config/opts are supposed to reside.

Reply:
this is like the old graph panel, it adds all its options to the base panel model

Reply:
OK. There's two obvious paths here:

1. If we can, as with the old graph panel, promise that the heatmap is deprecated and its schema will never change again, then we can safely (if still inelegantly) add it to the core dashboard schema, same as with #GraphPanel
2. If we can't make that promise, it implies we'd need to refactor heatmap to follow the same rules as other panel plugins, define a `models.cue` for it and confine any of its unique keys and values to the `options` and `fieldConfig.defaults` subtrees. We'd also need to write a TS migration that transforms existing heatmap panels into the new, rigorously-separated form in order for the schema system to be fully capable of handling it.

We also can combine these prior to GA of the schema system - embed a `#HeatmapPanel` definition in the core dashboard scuemata for now in order to get validation passing, then remove it once we've refactored heatmap to follow rigorous separation rules.

Also, to get out ahead of it, i've also seen `table-old` seems to add its options to the base panel, too - are there any others?